src/main/java/core/basesyntax/Calculator.java

```diff
@@ -0,0 +1,58 @@
+package core.basesyntax;
+
+public class Calculator {
+    private static final char ADDITION = '+';
+    private static final char SUBTRACTION = '-';
+    private static final char DIVISION = '/';
+    private static final char MULTIPLICATION = '*';
+    private static final char POWER = '^';
+
+    public double calculate(double a, double b, char operation) {
+        switch (operation) {
+            case ADDITION:
+                return add(a, b);
+            case SUBTRACTION:
+                return subtract(a, b);
+            case DIVISION:
+                return divide(a, b);
+            case MULTIPLICATION:
+                return multiply(a, b);
+            case POWER:
+                return raiseToPower(a, b);
+            default:
+                throw new IllegalArgumentException("Wrong operation");
+        }
+    }
+
+    private double add(double a, double b) {
+        return a + b;
+    }
+
+    private double subtract(double a, double b) {
+        return a - b;
+    }
+
+    private double divide(double a, double b) {
+        if (b == 0 || b == Double.MIN_VALUE) {
```

Comment:
```suggestion
        if (b == 0) {
```
I think this should not be an edge case